src/adm/settings.rs

```diff
@@ -176,12 +177,56 @@ where
 }
 
 #[derive(Debug, Default, Clone)]
-pub struct AdmSettings {
+pub struct AdmPse {
+    pub partner_id: String,
+    pub sub1: String,
+    pub endpoint: String,
+}
+
+/// ADM Partner/Sub1/Endpoint.
+/// These change depending on the type of device requesting the tile.
+impl AdmPse {
+    /// Return the information for a mobile connection
+    pub fn as_mobile(settings: &Settings) -> Self {
+        let default = Self::as_default(settings);
+        AdmPse {
```

Comment:
Instead of cloning the bits to create new `AdmPse`s every time we could construct them once at startup and store them on `ServerState`. I don't care much about these extra clones though so this as is is ok.